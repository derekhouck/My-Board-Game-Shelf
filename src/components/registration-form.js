import React from 'react';

export default function RegistrationForm(props) {
  return (
    <form className="registration-form">
      <label htmlFor="name">Name</label>
      <input type="text" name="name" />
      <label htmlFor="username">Username</label>
      <input type="text" name="username" />
      <label htmlFor="password">Password</label>
      <input type="password" name="password" id="password"/>
      <label htmlFor="passwordConfirm">Confirm password</label>
      <input type="password" name="passwordConfirm" id="passwordConfirm"/>
      <button type="submit">Create Account</button>
    </form>
  );
}