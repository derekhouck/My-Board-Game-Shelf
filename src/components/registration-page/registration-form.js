import React from 'react';
import { reduxForm, Field, focus } from 'redux-form';
import { registerUser } from '../../actions/users';
import { login } from '../../actions/auth';
import { nonEmpty, matches, length, isTrimmed, required, validateEmail } from '../../validators';

import Input from '../atoms/input';
import Button from '../atoms/button';

const passwordLength = length({ min: 8, max: 72 });
const matchesPassword = matches('password');


export class RegistrationForm extends React.Component {
  onSubmit(values) {
    const { email, name, password, username } = values;
    const user = { email, name, password, username };
    return this.props.dispatch(registerUser(user))
      .then(() => this.props.dispatch(login(username, password)));
  }

  render() {
    return (
      <form
        className="registration-form"
        onSubmit={this.props.handleSubmit(values => this.onSubmit(values))}>
        <fieldset>
          <legend>About you</legend>
          <Field
            component={Input}
            type="text"
            name="name"
            label="Name"
          />
          <Field
            component={Input}
            name="email"
            label="Email"
            type="email"
            validate={[required, validateEmail]}
          />
        </fieldset>
        <fieldset>
          <legend>Account details</legend>
          <Field
            component={Input}
            type="text"
            name="username"
            label="Username"
            validate={[required, nonEmpty, isTrimmed]}
          />
          <Field
            component={Input}
            type="password"
            name="password"
            id="password"
            label="Password"
            validate={[required, isTrimmed, passwordLength]}
          />
          <Field
            component={Input}
            type="password"
            name="passwordConfirm"
            id="passwordConfirm"
            label="Confirm password"
            validate={[required, matchesPassword, nonEmpty]}
          />
        </fieldset>
        <Button
          disabled={this.props.pristine || this.props.submitting}
          type="submit"
          label="Create Account"
          primary
        />
      </form>
    );
  }
}

export default reduxForm({
  form: 'registration',
  onSubmitFail: (errors, dispatch) => dispatch(focus('registration', Object.keys(errors)[0]))
})(RegistrationForm);