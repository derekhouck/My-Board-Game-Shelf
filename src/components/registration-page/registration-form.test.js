import React from 'react';
import { shallow } from 'enzyme';
import { RegistrationForm } from './registration-form';

describe('<RegistrationForm />', function () {
  it('renders without crashing', () => {
    const callback = jest.fn();
    const wrapper = shallow(<RegistrationForm handleSubmit={callback} />);
    expect(wrapper).toMatchSnapshot();
  });
});
