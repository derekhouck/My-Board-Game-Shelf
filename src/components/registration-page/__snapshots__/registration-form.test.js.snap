// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationForm /> renders without crashing 1`] = `ShallowWrapper {}`;
